app/controllers/organizations_controller.rb

```diff
@@ -110,10 +103,14 @@ def update
   end
 
   def delete
-    @organization = find_org_by_path params[:slug]
-    @organization.delete
-
-    redirect_to organizations_path( org_path: params[:org_path])
+    if @organization.allow_org_deletion?
+      @organization.delete 
+      
+      redirect_to organizations_path( org_path: params[:org_path])
+    else
+      @organization.errors.add('Cannot_delete', 'that organization has sub organizations')
+      render :edit
+    end
```

Comment:
i feel like this should have been done in the model. similar to [this solution](https://stackoverflow.com/questions/4054112/how-do-i-prevent-deletion-of-parent-if-it-has-child-records/10257516#10257516) however I have been unable to get `acts_as_nested_set` and `:dependent => :restrict_with_error` to play nicely. and when I tried to use `before_destroy` it didn't even seem to run anything I put in it.

Reply:
looking at the `acts_as_nested_set` it looks like the gem is really old and out of date. 

It may be worthwhile to look at a replacement.

Reply:
looks like you are actually using https://github.com/collectiveidea/awesome_nested_set which is more up to date